### PR TITLE
Add distributed implementation for recommender system

### PR DESCRIPTION
**Pull request by abhinavarora**
Fixes https://github.com/PaddlePaddle/Paddle/issues/7228

### CODE REVIEW

File: python/paddle/v2/fluid/tests/book_distribute/notest_recommender_system_dist.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+import paddle.v2.fluid.nets as nets
+from paddle.v2.fluid.optimizer import SGDOptimizer
+
+IS_SPARSE = True
+BATCH_SIZE = 256
+PASS_NUM = 100
+
+
+def get_usr_combined_features():
+
```
Comment (typhoonzero) [inline python/paddle/v2/fluid/tests/book_distribute/notest_recommender_system_dist.py]: Can you please remove these empty lines?

File: python/paddle/v2/fluid/tests/book_distribute/notest_recommender_system_dist.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+            'movie_title': 6,
+            'score': 7
+        }
+
+        for pass_id in range(PASS_NUM):
+            for data in train_reader():
+                outs = exe.run(trainer_prog,
+                               feed=func_feed(feeding, data, place),
+                               fetch_list=[cost])
+                out = np.array(outs[0])
```
Comment (gongweibao) [inline python/paddle/v2/fluid/tests/book_distribute/notest_recommender_system_dist.py]: Maybe we need to print some in the loop and before exit normally.

File: python/paddle/v2/fluid/tests/book_distribute/notest_recommender_system_dist.py (hunk 1 above)
Comment (abhinavarora) [reply python/paddle/v2/fluid/tests/book_distribute/notest_recommender_system_dist.py]: Thank you for pointing this out. Fixed in the latest commit.

File: python/paddle/v2/fluid/tests/book_distribute/notest_recommender_system_dist.py (hunk 2 above)
Comment (abhinavarora) [reply python/paddle/v2/fluid/tests/book_distribute/notest_recommender_system_dist.py]: Thank you. Added new print statements in the latest commit.

**Pull request merged**